pick.c

```diff
@@ -973,6 +983,17 @@ get_key(const char **key)
 		return CTRL_L;
 	}
 
+	if (buf[0] == '\033') {
+		struct pollfd	pfd;
+
+		pfd.fd = fileno(tty_in);
+		pfd.events = POLLIN;
+		if (poll(&pfd, 1, 200) == 0) {
+			vi_mode = !vi_mode;
```

Comment:
I would expect ESC to exit insert mode and not toggle Vi-mode.

```c
if (vi_mode)
  vi_mode = 0;
```

Reply:
... a separate key binding would then be necessary to enable the Vi-mode again ...

Reply:
The idea was some sort of "hybrid-mode":  the "insert mode" is the "emacs mode" with all the known key bindings. By pressing `ESC` you switch to the Vi command-mode and the keys are "translated" to appropriate function of the "emacs mode": some commands (or `ESC`) brings you back to the "emacs mode". The command line switch only activates the Vi command-mode at start to mimic the behavior of Vi: you have to press some command keys (like `a`, `i`, `/`) or `ESC` first to enter the search query ... don't know if Vi-addicts like or want this ... ;-)